compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoNop.cpp

```diff
@@ -84,4 +87,9 @@ createMaterializeEncodingIntoNopPass() {
   return std::make_unique<MaterializeEncodingIntoNopPass>();
 }
 
+void addEncodingToNopPasses(FunctionLikeNest &passManager) {
+  passManager.addPass(createMaterializeEncodingIntoNopPass)
+      .addPass(createCanonicalizerPass);
```

Comment:
I know what's happening, and I'm surprised that there are no failures in other backends. Anyway, so we need to add `.addPass(createBufferizeCopyOnlyDispatchesPass)` right after the `MaterializeEncodingIntoNop` pass.

The pass is doing early bufferization and create a linalg.copy ops (in generic op form) on memref types. It is for the `flow.tensor.store(flow.tensor.load)` cases. There are no compute ops at all, so we need to do the early bufferization and create the op on memrefs. It does not work on tensors because it's easily DCE-ed in tensor's world.

We usually address the issue in `addCommonTargetExecutablePreprocessingPasses`, and the pass does the conversion for us. However, it's not triggered in the PR because there is a `set_encoding` op in the middle. It becomes the form (i.e., `flow.tensor.store(flow.tensor.load))` after we run the nop pass. So I suggest to add the pass in the method, which generates the valid IR to later pipeline.

Reply:
It works magic. @hanhanW